### PR TITLE
cudnn conv group support

### PR DESCRIPTION
**Pull request by slayton58**
Enable the new convolution group functionality in cuDNN v7

### CODE REVIEW

File: caffe2/operators/conv_op_cudnn.cc [hunk 1]
```diff
@@ -462,6 +462,9 @@ bool CudnnConvOp::DoRunWithType() {
       CUDNN_ENFORCE(cudnnSetConvolutionMathType(
             conv_desc_, CUDNN_TENSOR_OP_MATH));
     }
+
+    // enable cuDNN conv groups
+    CUDNN_CHECK(cudnnSetConvolutionGroupCount(conv_desc_, group_));
```
Comment (Yangqing) [inline caffe2/operators/conv_op_cudnn.cc:467]: Do we need to guard this with ifdef as well?

Comment (lukeyeager) [reply caffe2/operators/conv_op_cudnn.cc:467]: It is guarded already (see above)

**Comment by 7oud** [comment]
@slayton58 Would the grouped convolution with cudnn 7 be faster than mannually convolution through groups? Could you give some test result?

**Comment by slayton58** [comment]
@7oud In best case special kernels that handle all groups in one shot will be used, worst case all groups in the convolution will be run in streams to try and acheive some overlap. I have no benchmarks handy, but I would strongly expect perf of the new routine to be `>=` the old loop over groups.

**Comment by sh1r0** [comment]
@slayton58 I found that I could not make the group_conv_test (caffe2/python/operator_test/group_conv_test.py) pass with cuda 8 and cudnn 7. Could you have a quick check about that?

**Comment by sh1r0** [comment]
The problem is caused by wrong dimension settings in cudnn tensor descriptors.

**Comment by slayton58** [comment]
@sh1r0 this is fixed in #1117

**Comment by sh1r0** [comment]
@slayton58 Good to know, thanks.

**Pull request closed without merging**